community[minor]: Add ability to specify file encoding and json encoding for FileChatMessageHistory

yabooung opened the pull request:
Description:
Add UTF-8 encoding support

Issue:
Inability to properly handle characters from certain languages (e.g., Korean)

Fix:
Implement UTF-8 encoding in FileChatMessageHistory

yabooung committed:
Add UTF-8 support

Description:
Add UTF-8 encoding support

Issue:
Inability to properly handle characters from certain languages (e.g., Korean)

Fix:
Implement UTF-8 encoding in FileChatMessageHistory

yabooung committed:
Merge branch 'master' into add-utf8-support-in-chat-message-histories

eyurtsev commented on libs/community/langchain_community/chat_message_histories/file.py:
```diff
@@ -20,21 +20,25 @@ def __init__(self, file_path: str) -> None:
         self.file_path = Path(file_path)
         if not self.file_path.exists():
             self.file_path.touch()
-            self.file_path.write_text(json.dumps([]))
```
write_text accepts an encoding parameter

write_text(json.dumps([]), encoding='utf-8')

eyurtsev commented on libs/community/langchain_community/chat_message_histories/file.py:
```diff
@@ -20,21 +20,25 @@ def __init__(self, file_path: str) -> None:
         if not self.file_path.exists():
             self.file_path.touch()
-            self.file_path.write_text(json.dumps([]))
+            with self.file_path.open("w", encoding="utf-8") as f:
```
To add the options in a non breaking way, you'll need to add encoding and ensure ascii as part of the initializer.

If we set the default encoding to utf-8, we could break existing user message history (e.g., if their default encoding is a windows encoding)

yabooung replied:
Thanks for the advice

eyurtsev reviewed (changes requested):
Thanks for the contribution! To merge, we need to expose the options via the initializer with defaults matching the previous defaults (encoding=None and ensure_ascii=True).

yabooung committed:
Merge branch 'master' into add-utf8-support-in-chat-message-histories

yabooung committed:
Merge branch 'langchain-ai:master' into add-utf8-support-in-chat-message-histories

yabooung committed:
Add encoding and ensure_ascii options

- Add encoding and ensure_ascii parameters to __init__ method
- Consistently apply encoding and ensure_ascii options throughout the class
- Maintain backward compatibility by preserving previous default values

eyurtsev commented on libs/community/langchain_community/chat_message_histories/file.py:
```diff
@@ -11,30 +11,48 @@
-    def __init__(self, file_path: str) -> None:
+    def __init__(
+        self,
+        file_path: str,
```
```suggestion
        file_path: str,
        *,
```

eyurtsev committed:
Add * operator

eyurtsev committed:
lint

yabooung committed:
Fix E501 lint error

- Resolved line length issue (E501) on line 25
- Shortened docstring to comply with 88 character limit

yabooung committed:
Merge branch 'master' into add-utf8-support-in-chat-message-histories

yabooung committed:
Fix line length
